src/snowflake/snowpark/_internal/analyzer/snowflake_plan.py

```diff
@@ -539,20 +529,17 @@ def join(
 
     def save_as_table(
         self,
-        table_name: Union[str, Iterable[str]],
+        table_name: str,
```

Comment:
The workaround given to customers is `save_as_table([db_name, schema_name, table_name])`. Will that still work if we revert this?

Reply:
Since the release is delayed, can we not revert and try to fix this bug for when `table_name` is `str`?

Reply:
Short version is we can't, to fix the `str` case, we have to go BCR route.

When `table_name` is `str`, it is not possible for us to tell whether the user intends `schema.table_name` or `table.name`. If we want to fix the case where we allow `table_name` with dots, we must change behavior such that 
- if users want to specify `schema.table_name`, they now use `save_as_table([schema, table_name])`
- if the users want to specify `table.name`, they use `save_as_table("table.name")`

For the above, we have to follow BCR process.

Reply:
^ I am wrong here. Looks like there is a way to decipher using quotes. For example, we allow something like `"schema.name"."table.name"` to allow dots in both schema name and table name.